build.cake

```diff
@@ -0,0 +1,108 @@
+#addin "Cake.FileHelpers"
+
+var target = Argument("target", "Default");
+var project = Argument("project", "*");
+
+var buildDir = "./build/";
+var sourceDir = "./exercises/";
+var projectDirs = GetFiles(buildDir + "**/" + project + ".csproj");
+
+var allProjects = GetFiles(buildDir + "**/*.csproj");
+
+/** The following exercises provide a stub implementation and the 
+    user has to complete the implementation. Therefore compiling 
+    these will result in failed compilation. Remove these and 
+    test the rest of the exercises.
+*/
+var defaultProjects = allProjects 
+    -  GetFiles(buildDir + "**/DotDsl.csproj")
+    -  GetFiles(buildDir + "**/Hangman.csproj")
+    -  GetFiles(buildDir + "**/React.csproj");
+
+/** These projects have full and working implementations, and have
+    all the tests enabled. These should pass without any changes.
+*/
+var refactoringProjects = GetFiles(buildDir + "**/TreeBuilding.csproj")
+    + GetFiles(buildDir + "**/Ledger.csproj")
+    + GetFiles(buildDir + "**/Markdown.csproj");
+
+
+Task("Clean")
+	.Does(() => {
+		CleanDirectory(buildDir);   
+    });
+
+
+// Copy everything to build so we make no changes in the actual files.
+Task("CopyExercises")
+    .IsDependentOn("Clean")
+    .Does(() => {
+        CopyDirectory(sourceDir, buildDir);
+    });
+
+
+Task("RestoreNugetPackages")
+    .IsDependentOn("CopyExercises")
+    .Does(() => {
+       foreach (var project in projectDirs) {
+           DotNetCoreRestore(project.ToString());
+       }
+    });
+
+
+Task("RestoreAndBuildStubImplementations")
+    .IsDependentOn("RestoreNugetPackages")
+    .Does(() => {
+        foreach (var project in defaultProjects) {
+            DotNetCoreBuild(project.ToString());
+        }
+});
+
+Task("EnableAllTests")
+    .IsDependentOn("RestoreAndBuildStubImplementations")
+    .Does(() => {
+        var testFiles = GetFiles(buildDir + "**/*Test.cs");
+        foreach (var testFile in testFiles) {
```

Comment:
This `foreach` can be removed since `ReplaceRegexInFiles` takes the glob as the param.